packages/cli/src/services/generator.ts

```diff
@@ -13,6 +19,16 @@ export async function generate<TInfo>(target: Target<TInfo>): Promise<void> {
   await fs.outputFile(renderPath, rendered)
 }
 
+export function template(name: string): string {
```

Comment:
This is great as it is, but one nice touch would be to have here a type that restricts the template names only to existing templates so a misuse is detected by the compiler.

One way is using an union type that restricts `string` to only valid values:

```suggestion
export function template(name: 'command' | 'event' | ...): string {
```

Another might be using the actual classes instead of strings:

```suggestion
export function template(name: Command | Event | ...): string {
```

But as I said, the current implementation is good too, we can make this improvement in future PRs too ;-)

Reply:
@javiertoledo Thanks for pointing that out! I fixed it by adding `TemplateType`.